api/app/signals/apps/api/serializers/nested/reporter.py

```diff
@@ -13,16 +13,20 @@ class Meta:
             'sharing_allowed',
         )
 
-    def to_representation(self, *args, **kwargs):
+    def to_representation(self, instance):
         """
         For backwards compatibility we replace null in email and phone fields with empty string.
         """
-        serialized = super().to_representation(*args, **kwargs)
+        serialized = super().to_representation(instance)
+
+        user = self.context['request'].user if 'request' in self.context else None
+        if not user.has_perm('signals.sia_can_view_contact_details'):
+            serialized['email'] = '*****' if serialized['email'] is not None else ''
```

Comment:
I think that the database contains both `""` and 'null' for missing email or phone numbers. In case `""` is used this will turn into `'*****'` suggesting contact details are present (but they are not).

The filter code contains a special case for this, see:
https://github.com/Amsterdam/signals/blob/master/api/app/signals/apps/api/filters/signal.py#L118

Reply:
Changed the if statements